mkosi.images/system/mkosi.sync

```diff
@@ -12,8 +12,21 @@ if [[ -d "$PKG_SUBDIR/.git" ]] && [[ "$(git -C "$PKG_SUBDIR" rev-parse HEAD)" ==
     exit 0
 fi
 
+# The repository on Salsa has the full upstream sources, so it's a waste of space to
+# redownload and duplicate everything, so do a sparse checkout as we only need the
+# packaging directory anyway
+if [[ "$DISTRIBUTION" = "debian" ]] || [[ "$DISTRIBUTION" = "ubuntu" ]]; then
+    sparse="--no-checkout --filter=tree:0"
+else
+    sparse=""
+fi
+
 if [[ ! -e "$PKG_SUBDIR" ]] || [[ -z "$(ls --almost-all "$PKG_SUBDIR")" ]]; then
-    git clone "$GIT_URL" --branch "$GIT_BRANCH" "$PKG_SUBDIR"
+    # shellcheck disable=SC2086
+    git clone "$GIT_URL" --branch "$GIT_BRANCH" $sparse "$PKG_SUBDIR"
+    if [[ "$DISTRIBUTION" = "debian" ]] || [[ "$DISTRIBUTION" = "ubuntu" ]]; then
+        git -C "$PKG_SUBDIR" sparse-checkout set --no-cone debian
```

Comment:
--no-cone is explicitly discouraged by the git sparse-checkout man page so I think we should figure out how to do this without it. I'm confident it's possible, it's just that this feature is rather obtuse just like the rest of git

Reply:
Reading the sparse checkout documentation, I think we can safely drop `--no-cone` here.

Reply:
Ah nvm I get the reason for `--no-cone` now. Since this is extremely non-trivial to figure out, can you please add a comment explaining why we don't use the recommended cone mode?

Reply:
I have actually no idea lol, these two options were in the first result for sparse checkout on Google XD

Reply:
cone mode will include the top level files in the repository as well, `--no-cone` will probably allow you to have only the `debian` directory and nothing else.